File: 0119-pascals-triangle-ii/0119-pascals-triangle-ii.java
class Solution {
    public List<Integer> getRow(int rowIndex) {
        List<List<Integer>> pyramid = new ArrayList<>();

        for (int i=0; i<rowIndex+1; i++){
            
            List<Integer> row = new ArrayList<>();

            row.add(1);

            for (int j=1; j<i; j++){
                row.add(pyramid.get(i-1).get(j) + pyramid.get(i-1).get(j-1));
            }

            if (i>0){
                row.add(1);   
            }

            pyramid.add(row);
        }

        return pyramid.get(rowIndex);
    }
}

File: 0119-pascals-triangle-ii/0119-pascals-triangle-ii.py
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:        
        pyramid = []
        for i in range(rowIndex+1):
            row = [1]
            if i > 0:
                for j in range(1,i):
                    row.append(pyramid[i-1][j] + pyramid[i-1][j-1])
                row.append(1)
            pyramid.append(row)
        
        return pyramid[rowIndex]